modules/calypso-redirects.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+if ( ! class_exists( 'Calypso_Redirects' ) ) :
+
+class Calypso_Redirects {
+
+	public function __construct() {
+		if ( ! get_option( 'jetpack_calypso_redirects_active' ) || ! is_admin() ) {
+			return;
+		}
+
+		add_action( 'admin_menu', array( $this, 'menu_redirects' ), 30 );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+
+		add_filter( 'allowed_redirect_hosts', array( $this, 'allow_calypso_domain' ), 10, 1 );
+
+		add_action( 'load-toplevel_page_calypso-plugins', array( $this, 'plugins_redirect' ) );
+
+		if ( current_user_can( 'list_users' ) ) {
+			$this->register_user_redirect_handlers();
+		}
+	}
+
+	private function register_user_redirects() {
+		remove_submenu_page( 'users.php', 'users.php' );
+		remove_submenu_page( 'users.php', 'user-new.php' );
+		remove_submenu_page( 'users.php', 'profile.php' );
+
+		// replace each menu item one by one with its redirect
+		add_submenu_page( 'users.php', __( 'All Users' ), __( 'All Users' ), 'list_users', 'calypso-users', array( $this, 'users_redirect' ) );
+		add_submenu_page( 'users.php', __( 'Add New' ), __( 'Add New' ), 'promote_users', 'calypso-users-new', array( $this, 'users_new_redirect' ) );
+		add_submenu_page( 'users.php', __( 'Your Profile' ), __( 'Your Profile' ), 'read', 'calypso-users-profile', array( $this, 'users_profile_redirect' ) );
+	}
+
+	private function register_user_redirect_handlers() {
+		add_action( 'load-users_page_calypso-users', array( $this, 'users_redirect' ) );
+		add_action( 'load-users_page_calypso-users-new', array( $this, 'users_new_redirect' ) );
+		add_action( 'load-users_page_calypso-users-profile', array( $this, 'users_profile_redirect' ) );
+		add_action( 'load-users_page_calypso-users-settings', array( $this, 'users_settings_redirect' ) );
+	}
+
+	private function register_track_event( $event ) {
+		// record in tracks
+		jetpack_require_lib( 'tracks/client' );
+		jetpack_tracks_record_event( wp_get_current_user(), $event );
+	}
+
+	public function allow_calypso_domain( $domains ) {
+		$domains[] = 'wordpress.com';
+
+		return $domains;
+	}
+
+	public function enqueue_scripts() {
+		wp_enqueue_style( 'calypso-redirects-css', plugins_url( 'calypso-redirects/calypso-redirects.css' , __FILE__ ), array(), '20170922-1' );
```

Comment:
I think we can use `JETPACK__VERSION` here for cache buster to avoid the need to update this manually in the future.